### DISCUSSION TITLE
Distributing tables in multicolumn lists

### DISCUSSION THREAD

**Discussion by psads-git** [typst/typst#4495] [Q&A] [answered]
Consider the following code below. I would like to obtain the following result:

- To minimize the waste of vertical space between table items in the list. (Notice that between the two upper table items and the two lower table items, there is a reasonable amount of vertical space that should be removed.

- The tables aligned with the item label (`a)`, `b)`, `c)`, ...)

- To avoid any text wrapping in the table cells.

- No page break between outer items. (In my example, there is a page break between item `1.` And item `2.`)

Thanks!

```
#set enum(full: true, numbering: (..n) => {
    let format = if n.pos().len() > 1 {"a)"} else {"1."}
    numbering(format, n.pos().last())
})

#let Q01 = [
Consider the following matrix:

$
mat(
  1, 2;
  4, 9;
)
$]

#let Q02 = [
Solve the following integral:

$
integral x e^(x) dif x.
$]

#let l = (Q02, Q01)

#let tbl = [#table(
  columns: 3,
  table.header(
    [Substance],
    [Subcritical °C],
    [Supercritical °C],
  ),
  [Hydrochloric Acid],
  [12.0], [92.1],
  [Sodium Myreth Sulfate],
  [16.6], [104],
  [Potassium Hydroxide],
  table.cell(colspan: 2)[24.7],
)]

#for i in range(l.len()) {
  [+ #l.at(i)
      #block(
        columns(2, gutter: 11pt)[
       + #scale(x:80%, y: 80%, 
       [ #tbl])
       + #tbl #colbreak() 
       + #scale(x:80%, y: 80%, 
       [ #tbl])
       + #tbl]
      )
  ]
}
```

![image](https://github.com/typst/typst/assets/75945439/9ed1c6f0-ff83-4cea-ad70-05fb046f683d)

**Comment by TS60** [comment]
I think, the most simple solution would be to put all things again in one table. Then you could place the tables within this table as wanted. A pagebreak can be prevented using `box`.

**Reply by laurmaedje** [reply, marked as answer]
Two comments: 
- Consider using a grid instead of a table if it's just for layout
- A `block(breakable: false)` is preferable over a box. The main purpose of a box is to put things into a paragraph.
